src/requests/base/HTTPGetRequest.ts

```diff
@@ -132,14 +139,18 @@ export default abstract class HTTPGetRequest extends HTTPRequest {
         }
     }
 
-    protected sendResponse(documents: Object[] | Object) {
+    protected sendResponse(documents: NonNullable<unknown>) {
         const filterString: string = Utils.generateFilterString(
             this.validRequestQueryParameters,
             this._req,
         );
-        const nestedArray = !Array.isArray(documents)
-            ? Object.values(documents).find((value) => Array.isArray(value))
-            : [];
+        const nestedArray = (
+            !Array.isArray(documents)
+                ? Object.values(documents).find((value) => Array.isArray(value))
+                : []
+        ) as { length: number };
+        // todo: ^ there has to be a sane way to do that
```

Comment:
Yeah this part of the code is really gross. Not sure how to make it work for all kinds of responses though.